src/Acuminator/Acuminator.Utilities/CodeAnalysisSettings.cs

```diff
@@ -11,17 +11,51 @@ namespace Acuminator.Utilities
 	public class CodeAnalysisSettings
 	{
 		public static CodeAnalysisSettings Default => 
-			new CodeAnalysisSettings(recursiveAnalysisEnabled: true);
+			new CodeAnalysisSettings(
+				recursiveAnalysisEnabled: true, 
+				isvSpecificAnalyzersEnabled: false);
 
 		public CodeAnalysisSettings()
 		{
 		}
 
-		private CodeAnalysisSettings(bool recursiveAnalysisEnabled)
+		private CodeAnalysisSettings(bool recursiveAnalysisEnabled, bool isvSpecificAnalyzersEnabled)
 		{
 			RecursiveAnalysisEnabled = recursiveAnalysisEnabled;
+			IsvSpecificAnalyzersEnabled = isvSpecificAnalyzersEnabled;
 		}
 
 		public virtual bool RecursiveAnalysisEnabled { get; }
+		public virtual bool IsvSpecificAnalyzersEnabled { get; }
+
+		public CodeAnalysisSettings WithRecursiveAnalysisEnabled()
+		{
+			return WithRecursiveAnalysisEnabledValue(true);
+		}
+
+		public CodeAnalysisSettings WithRecursiveAnalysisDisabled()
+		{
+			return WithRecursiveAnalysisEnabledValue(false);
+		}
+
+		public CodeAnalysisSettings WithIsvSpecificAnalyzersEnabled()
+		{
+			return WithIsvSpecificAnalyzersEnabledValue(true);
+		}
+
+		public CodeAnalysisSettings WithIsvSpecificAnalyzersDisabled()
+		{
+			return WithIsvSpecificAnalyzersEnabledValue(false);
+		}
+
+		protected virtual CodeAnalysisSettings WithRecursiveAnalysisEnabledValue(bool value)
```

Comment:
@dropsonic are the othet methods really needed? You can just make this protected methods public

Reply:
@SENya1990 yes, but it will decrease readability

Reply:
not so much  - compare `CodeAnalysisSettings.Default.WithIsvSpecificAnalyzersEnabledValue(false)` with
`CodeAnalysisSettings.Default.WithRecursiveAnalysisDisabled()`
I agree that second option is more readable but should we need to add three methods in the future for every new setting? The first option is quite clear too.

Reply:
We rarely add new options, so it isn't a big deal.

Reply:
Ok then